### ISSUE TITLE
Tag/Content attributes should silently ignore unkown attributes

### ISSUE THREAD

**Issue by daiheitan** [serde-rs/serde#816]
Currently if I use `#[serde(tag, content)]` on a tagged enum, it seems that serde would throw error when the JSON string contains attributes other than the tag and the content. I think silently ignore the unknown field could be better?

**Comment by dtolnay** [comment]
Really we should respect deny_unknown_fields.

```rust
// as you said
#[derive(Serialize, Deserialize)]
#[serde(tag = "t", content = "c")]
enum E { /* ... */ }

// current behavior
#[derive(Serialize, Deserialize)]
#[serde(tag = "t", content = "c", deny_unknown_fields)]
enum F { /* ... */ }
```

I would welcome a PR if you have time to work on this.

**Comment by daiheitan** [comment]
Seems to have to modify quite a lot of logic to make this working. Currently only Structs are allowed to use `deny_unknown_fields`, as enums are variants, they could not use the field ignore logic when deserializing.

**Comment by daiheitan** [comment]
ping @dtolnay 
Any ideas?

**Comment by dtolnay** [comment]
This is not a high priority for me but I have scheduled it as something I intend to address in the next major release if a PR has not been submitted by the time I get to it.

**Comment by daiheitan** [comment]
Thanks, I'll wait

**Comment by dtolnay** [comment]
@daiheitan this has been fixed in serde 1.0.2.

**Comment by daiheitan** [comment]
Thanks, that's great!
